java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -179,12 +180,22 @@ public String toString() {
 
   public ProgramState addConstraint(SymbolicValue symbolicValue, Constraint constraint) {
     PMap<SymbolicValue, Constraint> newConstraints = constraints.put(symbolicValue, constraint);
-    if (newConstraints != constraints) {
+    if (newConstraints != constraints && !isSymbolicValueOfVolatileField(symbolicValue)) {
```

Comment:
Wondering if it wouldn't not be better to test for "volatileness" of SV prior to adding the constraints.